### PR TITLE
e2e: skip when 'etcdProcess' is nil

### PR DESCRIPTION
**Pull request by gyuho**
For https://github.com/coreos/etcd/issues/6561.

### CODE REVIEW

File: e2e/etcd_test.go [hunk 1]
```diff
@@ -424,6 +424,9 @@ func (epc *etcdProcessCluster) StopAll() (err error) {
 func (epc *etcdProcessCluster) Close() error {
 	err := epc.StopAll()
 	for _, p := range epc.procs {
+		if p == nil {
```
Comment (xiang90) [inline e2e/etcd_test.go]: when will p equal to nil? add a comment?

Comment (gyuho) [inline e2e/etcd_test.go]: Just added comment. PTAL. Thanks.

File: e2e/etcd_test.go [hunk 2]
```diff
@@ -424,6 +424,9 @@ func (epc *etcdProcessCluster) StopAll() (err error) {
 func (epc *etcdProcessCluster) Close() error {
 	err := epc.StopAll()
 	for _, p := range epc.procs {
+		if p == nil { // newEtcdProcess failed in the middle
```
Comment (xiang90) [inline e2e/etcd_test.go]: if it fails in the middle, the test should already fail. this only helps the test to clean it up after failure. right?

Comment (gyuho) [inline e2e/etcd_test.go]: Yeah... This only helps cleaning without nil pointer reference. It makes `Close` consistent with `StopAll` method though.

Comment (xiang90) [inline e2e/etcd_test.go]: OK. So this does not actually fix #6561, right?

Comment (xiang90) [inline e2e/etcd_test.go]: Can we add more comments? mention that this is for cleaning up tests when there is a failure in the middle?

Comment (gyuho) [inline e2e/etcd_test.go]: Just added more comment, and changed commit body to not close that issue. PTAL. Thanks.

**Comment by xiang90** [comment]
LGTM

**Pull request merged**